libs/blocks/modal/decorators/langnav.js

```diff
@@ -0,0 +1,37 @@
+import { getConfig } from '../../../utils/utils.js';
+
+function handleEvent(prefix, link, config) {
+  document.cookie = `international=${prefix};path=/`;
+  sessionStorage.setItem('international', prefix);
+  fetch(link.href, { method: 'HEAD' }).then((resp) => {
+    if (!resp.ok) throw new Error('request failed');
+    window.location.assign(link.href);
+  }).catch(() => {
+    const prefixUrl = prefix ? `/${prefix}` : '';
+    window.location.assign(`${prefixUrl}${config.contentRoot || ''}/`);
+  });
+}
+
+function decorateLink(link, config, path) {
+  let pathname = link.getAttribute('href');
+  try { pathname = new URL(pathname).pathname; } catch (e) { /* href does not contain domain */ }
+  const linkParts = pathname.split('/');
+  const prefix = linkParts[1] || 'us';
```

Comment:
If the path is `/acrobat/page/to/something.html`, this will assign a prefix of `acrobat`

Reply:
I'm not expecting links in this format. Links should technically be arriving in this format: 
domain/de/
OR
/de/
OR
/

If links arrive in formats other than this, then I'll probably need to rethink my whole approach tbh. Can you think of a test case where links wouldn't have this format? I've tested it on [milo](https://main--milo--adobecom.hlx.page/?milolibs=local#langnav), [bacom](https://main--bacom--adobecom.hlx.page/customer-success-stories?martech=off&milolibs=local#langnav) and a [private project](https://main--locale-testing--vhargrave.hlx.page/milo-coffee/?milolibs=local#langnav) so far.